softlayer/resource_softlayer_lb_vpx_service.go

```diff
@@ -202,14 +217,34 @@ func resourceSoftLayerLbVpxServiceUpdate(d *schema.ResourceData, meta interface{
 		template.ConnectionLimit = sl.Int(data.(int))
 	}
 
-	_, err = service.Id(nadcId).UpdateLiveLoadBalancer(&datatypes.Network_LoadBalancer_VirtualIpAddress{
-		Name:     sl.String(vipName),
-		Services: []datatypes.Network_LoadBalancer_Service{template},
-	})
+	lbVip := &datatypes.Network_LoadBalancer_VirtualIpAddress{
+		Name: sl.String(vipName),
+		Services: []datatypes.Network_LoadBalancer_Service{
+			template},
+	}
+
+	successFlag := true
+	for count := 0; count < 10; count++ {
+		successFlag, err = service.Id(nadcId).UpdateLiveLoadBalancer(lbVip)
+		log.Printf("[INFO] Updating Loadbalancer service %s successFlag : %t", serviceName, successFlag)
+
+		if err != nil && strings.Contains(err.Error(), "Operation already in progress") {
+			log.Printf("[INFO] Updating Loadbalancer service %s Error : %s. Retry in 10 secs", serviceName, err.Error())
+			time.Sleep(time.Second * 10)
+			continue
+		}
+
+		break
+	}
```

Comment:
this block of code seems to be the same as in the Create function. If so, it would be good to refactor into its own function and reuse from both places.